### PR TITLE
Respect max-requests in batch-deletes for DynamoDB

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

File: versioned/persist/dynamodb/src/main/java/org/projectnessie/versioned/persist/dynamodb/DynamoDatabaseAdapter.java [hunk 1]
```diff
@@ -288,33 +288,47 @@ protected void doCleanUpCommitCas(
...
+                  .deleteRequest(
+                      b ->
+                          b.key(
+                              Collections.singletonMap(
+                                  KEY_NAME,
+                                  AttributeValue.builder().s(keyPrefix + hash.asString()).build())))
+                  .build());
+      requests++;
+
+      if (requests == DYNAMO_BATCH_WRITE_MAX_REQUESTS) {
```
Comment (tomekl007) [inline versioned/persist/dynamodb/src/main/java/org/projectnessie/versioned/persist/dynamodb/DynamoDatabaseAdapter.java:317]: can we add a test for it?

File: versioned/persist/dynamodb/src/main/java/org/projectnessie/versioned/persist/dynamodb/DynamoDatabaseAdapter.java [hunk 2]
```diff
@@ -288,33 +288,47 @@ protected void doCleanUpCommitCas(
...
+      }
+    }
+
+    // close() is actually a flush, implementing AutoCloseable for easier use of BatchDelete using
+    // try-with-resources.
+    @Override
+    public void close() {
+      if (requests > 0) {
+        client.client.batchWriteItem(b -> b.requestItems(requestItems));
+        requestItems.clear();
```
Comment (tomekl007) [inline versioned/persist/dynamodb/src/main/java/org/projectnessie/versioned/persist/dynamodb/DynamoDatabaseAdapter.java:328]: what happens to `requestItems` that are still in the buffer but not written?

Comment (snazy) [reply versioned/persist/dynamodb/src/main/java/org/projectnessie/versioned/persist/dynamodb/DynamoDatabaseAdapter.java:328]: It's an `AutoCloseable`, right?

File: versioned/persist/dynamodb/src/main/java/org/projectnessie/versioned/persist/dynamodb/DynamoDatabaseAdapter.java (hunk 1 above)
Comment (snazy) [reply versioned/persist/dynamodb/src/main/java/org/projectnessie/versioned/persist/dynamodb/DynamoDatabaseAdapter.java:317]: will do

File: versioned/persist/dynamodb/src/main/java/org/projectnessie/versioned/persist/dynamodb/DynamoDatabaseAdapter.java (hunk 2 above)
Comment (tomekl007) [reply versioned/persist/dynamodb/src/main/java/org/projectnessie/versioned/persist/dynamodb/DynamoDatabaseAdapter.java:328]: Yes, but it seems that we are ignoring additional records (`> DYNAMO_BATCH_WRITE_MAX_REQUESTS`).
Or is the `batchWriteItem` doing the write of the remaining items? If yes, we will still write `> DYNAMO_BATCH_WRITE_MAX_REQUESTS` records. Is it something that we expect?

Comment (snazy) [reply versioned/persist/dynamodb/src/main/java/org/projectnessie/versioned/persist/dynamodb/DynamoDatabaseAdapter.java:328]: `batchWriteItem` is the Dynamo-API call

**Review by tomekl007 — APPROVED** [review]

**Pull request merged**